app/models/reviewable.rb

```diff
@@ -766,8 +782,9 @@ def self.find_by_flagger_or_queued_post_creator(id:, user_id:)
     )
   end
 
-  def self.unknown_types
-    Reviewable.pending.distinct.pluck(:type) - Reviewable.sti_names
+  def self.unknown_types_and_sources
+    Reviewable.pending.distinct.pluck(:type, :type_source) -
+      Reviewable.sti_names.map { |n| [n, Reviewable.source_for(n)] }
```

Comment:
Small perf nit but it looks like this line can be calculated once and then memoized.